File: inhibition_analysis/gene_amplification.R
#myc gene copy number dataset
#pre-processed in Python
myc_cn <- read.csv('C:/Users/james/Desktop/Sumoylation_Analysis/project/inhibition_analysis/myc_copy_number.csv', fileEncoding="UTF-8-BOM") 
inhibition <- read.csv("inhibition_data.csv", fileEncoding="UTF-8-BOM") #inhibition data 

#now process inhibition data by eliminating all spaces and special characters
inhibition$cell <- gsub(" ", "", inhibition$cell)
inhibition$cell <- gsub("-", "", inhibition$cell)
inhibition$cell <- gsub("\\.", "", inhibition$cell) #to remove dot, add 2 backslashes
inhibition$cell <- toupper(inhibition$cell)

inhibition

myc_cn_AUC <- merge(x = myc_cn, 
                y = inhibition,
                by = "cell")

distinct(myc_cn_AUC, cell)

#drop duplicate column "type"
myc_cn_AUC <- myc_cn_AUC[, !names(myc_cn_AUC) %in% c("type")]

head(myc_cn_AUC)

mean(myc_cn_AUC$MYC)
min(myc_cn_AUC$MYC)
max(myc_cn_AUC$MYC)

#Set different thresholds for MYC copy number and observe the mean AUC values. 
above_3cn_myc <- myc_cn_AUC[myc_cn_AUC$MYC > 3, ]
#there is only 1 observation

above_2cn_myc <- myc_cn_AUC[myc_cn_AUC$MYC >= 2, ]
below_2cn_myc <- myc_cn_AUC[myc_cn_AUC$MYC < 2, ]
mean(above_2cn_myc$AUC)
mean(below_2cn_myc$AUC)

File: inhibition_analysis/gene_amplification.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

#import copy number dataset and cell line info dataset
cn = pd.read_csv('C:/Users/james/Desktop/Sumoylation_Analysis/data/CCLE_gene_cn.csv')
info = pd.read_csv('C:/Users/james/Desktop/Sumoylation_Analysis/data/sample_info.csv')
cn.head()
info.head()

#change the column names and subset desired columns
cn = cn.rename(columns = {'Unnamed: 0':'ID'})
cn = cn.rename(columns = {'MYC (4609)':'MYC'})
myc_df = cn[['ID', 'MYC']]
myc_df.head()

info = info.rename(columns = {'DepMap_ID':'ID'})
info = info[['ID', 'lineage', 'stripped_cell_line_name']]
info = info.rename(columns = {'stripped_cell_line_name':'cell'})
info.head()

myc_cn = info.merge(myc_df, on='ID')
myc_cn.head()
myc_cn['MYC'].describe()

#export myc copy number dataset to csv
myc_cn.to_csv("C:/Users/james/Desktop/Sumoylation_Analysis/project/inhibition_analysis/myc_copy_number.csv")

#import inhibition dataset
inhibition = pd.read_csv("C:/Users/james/Desktop/Sumoylation_Analysis/project/inhibition_analysis/inhibition_data.csv")
inhibition = inhibition[['cell', 'AUC']]

#delete all whitespace
inhibition['cell'] = inhibition['cell'].str.replace(' ', '')
#capitalize all letters
inhibition['cell'] = inhibition['cell'].str.upper()
inhibition.head()

final_df = pd.merge(myc_cn, inhibition, on='cell')
final_df.head()
final_df.describe()


#plot AUC vs MYC Copy number
MYC_plot = sns.scatterplot(data=final_df, x="AUC", y="MYC", color='red')
MYC_plot.set(xlabel='AUC', ylabel='MYC copy number in log2(x+1)')

#plt.show()
plt.savefig('C:/Users/james/Desktop/Sumoylation_Analysis/project/inhibition_analysis/MYC_plot.png')
